### ISSUE TITLE
Proper version tags for docker images

### ISSUE THREAD

**Issue by mazay** [hatamiarash7/Mikrotik-Exporter#19]
Currently, docker images are tagged as `latest` or with architecture type. Using the `latest` tag isn't a good idea as you lose control over your infra - on a container restart (especially if running on k8s) you might get a newer version of the application with braking changes and/or get unexpected results.

**Comment by mazay** [comment]
https://github.com/marketplace/actions/docker-setup-buildx - might be helpful

**Comment by hatamiarash7** [comment]
Yes, you are right about that. There were problems when developing the automatic release system that I could not use the `buildx` command ( many errors in Github Actions ). So I had to use the old methods like `docker manifest` as you can see here :

https://github.com/hatamiarash7/Mikrotik-Exporter/blob/a9db65970b4be4b882b1f4b32f11f517ff49bab6/.github/workflows/go.yml#L94-L108

Since then, I have not updated the automatic publishing system, but I will definitely put it in the to-do list because I also notice version problems.

I would also be happy if you could participate in this work.   
Thank you

**Comment by mazay** [comment]
Hi @hatamiarash7 , I've been looking through the pipeline and have a couple of changes I'd like to introduce, which I'd like to check with you first.

1. since the pipeline is always run on pushes to master it's not easy to distinguish regular pushes from releases, thus I'd like to introduce new trigger:
```yaml
on:
  release:
    types:
      - published
```

This was we could use either release tag or branch name for the image tag.

2. I'd suggest moving most of the build logic to `Makefile`, this will simplify building stuff locally without the need to push

So, WDYT?
